src/zombies.test.ts

```diff
@@ -1,30 +1,121 @@
+  return typeof num === 'number' && Number.isInteger(num) && num >= 0;
+};
+
+const isValidString = (name: string): name is string => {
+  return typeof name === 'string' && name.length > 0;
+};
+
+type ZombieReturnObj = {
+  isFull: () => boolean;
+  addZombie: (zombie: string) => void;
+  zombiesInRoom: () => string[];
+  noZombiesInRoom: () => number;
+  spaceLeft: () => number; 
+};
+
 const createRoom = (capacity: number) => {
-  const _capacity = capacity;
+  if (!numIsValidInteger(capacity)) return `Error: Parameter 'capacity' must be a positive integer`;
+  
+  const _capacity = capacity; 
+  let _zombiesInRoom: string[] = [];
 
   return {
-    isFull: () => true,
+    isFull: () => {
+      if (_capacity === 0 || _capacity === _zombiesInRoom.length) return true;      
+      
+      return false;
+    },
+    addZombie: (zombie: string) => {
+      if (isValidString(zombie)) {
+        if (_capacity === _zombiesInRoom.length) {
+          _zombiesInRoom.splice(0, 1, zombie);
+        } else {
+          _zombiesInRoom.push(zombie);
+        }  
+      } else {
+        console.log(`*** No zombie was added to the room since you tried to add "${zombie}". ***`);
+      }          
+    },
+    zombiesInRoom: () => _zombiesInRoom,
+    noZombiesInRoom: () => _zombiesInRoom.length, 
+    spaceLeft: () => capacity - _zombiesInRoom.length,
   };
 };
 
-test("room is not full", () => {
-  const room = createRoom(0);
+test("room is full", () => {
+  const room = createRoom(0) as ZombieReturnObj;
+
+  const result = room.isFull();  
+
+  ok(result);
+});
+
+test("empty room that fits one zombie is not full", () => {
+  const room = createRoom(1) as ZombieReturnObj;
+
+  const result = room.isFull();
+
+  ok(!result);
+});
+
+test("empty room cannot fit any zombies", () => {
+  const room = createRoom(0) as ZombieReturnObj;
 
   const result = room.isFull();
 
   ok(result);
 });
 
-test.skip("empty room that fits one zombie is not full", () => {});
+test("one-roomer becomes full when a zombie is added", () => {
+  const room = createRoom(1) as ZombieReturnObj;
 
-test.skip("empty room cannot fit any zombies", () => {});
+  room.addZombie('Ugh Lee');
 
-test.skip("one-roomer becomes full when a zombie is added", () => {});
+  const result = room.isFull();  
 
-test.skip("two-roomer is not full when a zombie is added", () => {});
+  ok(result);
+});
+
+test("two-roomer is not full when a zombie is added", () => {
+  const room = createRoom(2) as ZombieReturnObj;
+
+  room.addZombie('Ugh Lee');
+
+  const result = room.isFull();  
+
+  ok(!result);
+});
+
+test("second zombie consumes first zombie when added to a one-roomer", () => {
+  const room = createRoom(1) as ZombieReturnObj;
 
-test.skip("second zombie consumes first zombie when added to a one-roomer", () => {});
+  room.addZombie('Ugh Lee');
+  room.addZombie('Bloody Mary');
 
-// You are free to add more tests that you think are relevant!
+  const zombieInRoom = room.zombiesInRoom()[0];
```

Comment:
Try to not cherry-pick elements from the data structure like this. Just look at the entire list. Otherwise you won't be testing a whole unit, but a partial unit.